pkg/resource_builders/envoyconfig/listeners.go

```diff
@@ -0,0 +1,260 @@
+	envoy_extensions_access_loggers_file_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/access_loggers/file/v3"
+	envoy_extensions_filters_http_ratelimit_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/ratelimit/v3"
+	envoy_extensions_filters_listener_proxy_protocol_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/listener/proxy_protocol/v3"
+	http_connection_manager_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	envoy_extensions_transport_sockets_tls_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+func ListenerHTTP_v1(desc envoyDynamicConfigDescriptor) (envoy.Resource, error) {
+	opts := desc.(*saasv1alpha1.ListenerHttp)
+
+	listener := &envoy_config_listener_v3.Listener{
+		Name:            desc.GetName(),
+		Address:         Address_v1("0.0.0.0", opts.Port),
+		ListenerFilters: ListenerFilters_v1(opts.CertificateSecretName != nil),
+		FilterChains: []*envoy_config_listener_v3.FilterChain{{
+			Filters: []*envoy_config_listener_v3.Filter{{
+				Name: "envoy.filters.network.http_connection_manager",
+				ConfigType: &envoy_config_listener_v3.Filter_TypedConfig{
+					TypedConfig: func() *anypb.Any {
+						any, err := anypb.New(
+							&http_connection_manager_v3.HttpConnectionManager{
+								AccessLog: AccessLogConfig_v1(desc.GetName(), opts.CertificateSecretName != nil),
+								CommonHttpProtocolOptions: func() *envoy_config_core_v3.HttpProtocolOptions {
+									po := &envoy_config_core_v3.HttpProtocolOptions{
+										IdleTimeout: durationpb.New(3600 * time.Second),
+									}
+									if opts.MaxConnectionDuration != nil {
+										po.MaxConnectionDuration = durationpb.New(opts.MaxConnectionDuration.Duration)
+									}
+									if opts.AllowHeadersWithUnderscores != nil && *opts.AllowHeadersWithUnderscores {
+										po.HeadersWithUnderscoresAction = envoy_config_core_v3.HttpProtocolOptions_ALLOW
+									} else {
+										po.HeadersWithUnderscoresAction = envoy_config_core_v3.HttpProtocolOptions_REJECT_REQUEST
+									}
+									return po
+								}(),
+
+								HttpFilters: HttpFilters_v1(opts.RateLimitOptions),
+								HttpProtocolOptions: func() *envoy_config_core_v3.Http1ProtocolOptions {
+									if opts.DefaultHostForHttp10 != nil {
+										return &envoy_config_core_v3.Http1ProtocolOptions{
+											AcceptHttp_10:         true,
+											DefaultHostForHttp_10: *opts.DefaultHostForHttp10,
+										}
+									}
+									return &envoy_config_core_v3.Http1ProtocolOptions{}
+								}(),
+								Http2ProtocolOptions: &envoy_config_core_v3.Http2ProtocolOptions{
```

Comment:
we have a case in backend http-internal listener we don't set this, is this a problem having it defined now by default? or it is totally harmless?

Reply:
It does not affect if http2 is not enabled, and there is an specific flag to enable http2.